src/core/thread/radio_selector.cpp

```diff
@@ -323,6 +323,13 @@ Mac::TxFrame &RadioSelector::SelectRadio(Message &aMessage, const Mac::Address &
 #endif
 
 exit:
+#if OPENTHREAD_CONFIG_RADIO_LINK_TREL_ENABLE
+    if (!Get<Trel::Interface>().IsEnabled())
+    {
+        selections.Remove(Mac::kRadioTypeTrel);
+    }
+#endif
```

Comment:
If the trel is selected, `Links::Send` will copy the message, even though it already knows trel is not started. This is an unnecessary overhead. Just like a socket tries binding a network interface which is not up, should it fail when upper layer explicitly selects a link which is not up now? Upper layers may select all or leave it unspecific, and allows Mac/SubMac layer to modify the selections.

Reply:
The current code in this PR as is (i.e. removing radio link from `selections`)  will cause issues and bugs (I pointed one case above). 

If you want to optimize this, it requires more thoughts and a different solution.

I proposed some solutions if we want to avoid `Aborted` error log (which I expect/assume is harmless other than polluting the logs and potentially confusing folks).

-----

On avoiding overhead:
- If we build OT stack with `RADIO_LINK_TREL_ENABLE`, we should then try to use it with trel enabled. 
- If we  then disable it at run-time, we should be ok with overhead and inefficiencies it may cause.
- The API to enable/disable trel is mainly intended as a way of testing multi-radio and trel.
- I assume it would be possible to add code to optimize all of this but to me it is not a worth-while effort and there are risks of getting it wrong.

Reply:
The main motivation was to avoid the confusing logs. I'm fine to leave it as is.